Cinema/src/test/java/com/example/cinema/service/MovieServiceTest.java

```diff
@@ -0,0 +1,97 @@
+package com.example.cinema.service;
+
+import com.example.cinema.exception.WrongGenreNameException;
+import com.example.cinema.mapper.MoviesMapper;
+import com.example.cinema.model.dbModel.MovieDB;
+import com.example.cinema.model.dbModel.ProjectionDB;
+import com.example.cinema.model.requestModel.MovieUpdateReq;
+import org.junit.Assert;
+import org.junit.jupiter.api.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.time.LocalTime;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.mockito.Mockito.*;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+class MovieServiceTest {
+
+    @Autowired
+    MoviesService moviesService;
+
+    @MockBean
+    MoviesMapper moviesMapper;
+
+    @MockBean
+    GenreService genreService;
+
+    @Test
+    void findTime() {
```

Comment:
Add edge case coverage for findTime, findAll, insert, update, getByName.

Reply:
Created